dandi/tests/test_metadata.py

```diff
@@ -359,3 +360,283 @@ def test_species():
+                "measurementTechnique": ["behavioral technique"],
+                "variableMeasured": ["EyeTracking"],
+            },
+        ),
+        (
+            ["CompassDirection"],
+            {
+                "approach": ["behavioral approach"],
+                "measurementTechnique": ["behavioral technique"],
+                "variableMeasured": ["CompassDirection"],
+            },
+        ),
+        (
+            ["ProcessingModule"],
+            {
+                "approach": None,
+                "measurementTechnique": ["analytical technique"],
+                "variableMeasured": ["ProcessingModule"],
+            },
+        ),
+        (
+            ["RGBImage"],
+            {
+                "approach": None,
+                "measurementTechnique": ["photographic technique"],
+                "variableMeasured": ["RGBImage"],
+            },
+        ),
+        (
+            ["DecompositionSeries"],
+            {
+                "approach": None,
+                "measurementTechnique": ["fourier analysis technique"],
+                "variableMeasured": ["DecompositionSeries"],
+            },
+        ),
+        (
+            ["Units"],
+            {
+                "approach": ["electrophysiological approach"],
+                "measurementTechnique": ["spike sorting technique"],
+                "variableMeasured": ["Units"],
+            },
+        ),
+        (
+            ["Spectrum"],
+            {
+                "approach": None,
+                "measurementTechnique": ["fourier analysis technique"],
+                "variableMeasured": ["Spectrum"],
+            },
+        ),
+        (
+            ["OptogeneticStimulusSIte"],
+            {
+                "approach": ["optogenetic approach"],
+                "measurementTechnique": None,
+                "variableMeasured": ["OptogeneticStimulusSIte"],
+            },
+        ),
+        (
+            ["OptogeneticSeries"],
+            {
+                "approach": ["optogenetic approach"],
+                "measurementTechnique": None,
+                "variableMeasured": ["OptogeneticSeries"],
+            },
+        ),
+        (
+            ["CurrentClampSeries (94)"],
+            {
+                "approach": ["electrophysiological approach"],
+                "measurementTechnique": ["current clamp technique"],
+                "variableMeasured": ["CurrentClampSeries"],
+            },
+        ),
+    ],
+)
+def test_ndtypes(ndtypes, asset_dict):
+    asset = BareAssetMeta(
+        contentSize=1,
+        encodingFormat="application/x-nwb",
+        digest={"dandi:dandi-etag": "0" * 32 + "-1"},
+        path="test.nwb",
+    )
+    asset = process_ndtypes(asset, ndtypes)
+    for key in ["approach", "measurementTechnique"]:
+        if asset_dict.get(key) is None:
+            assert getattr(asset, key) == []
+        else:
+            assert getattr(getattr(asset, key)[0], "name") == asset_dict.get(key)[0]
```

Comment:
Instead of `getattr(getattr(asset, key)[0], "name")`, just write `getattr(asset, key)[0].name`, and likewise for line 642.